lightning/src/ln/channel.rs

```diff
@@ -1682,6 +1682,10 @@ pub(super) trait InteractivelyFunded<SP: Deref> where SP::Target: SignerProvider
 
 	fn interactive_tx_constructor_mut(&mut self) -> &mut Option<InteractiveTxConstructor>;
 
+	fn interactive_tx_signing_session_mut(&mut self) -> &mut Option<InteractiveTxSigningSession>;
+
+	fn interactive_tx_signing_session_with_context_mut(&mut self) -> (&mut Option<InteractiveTxSigningSession>, &mut ChannelContext<SP>);
```

Comment:
Why is it important to have both returned in one call? Could two calls to `interactive_tx_signing_session_mut` and `context_mut` achieve the same?

Reply:
They both borrow mutable references, so the borrow checker will complain. Though maybe adding two lifetimes to the trait would get around it. Will give it a try.